### PR TITLE
move app folder to addon folder and add reexports

### PR DESCRIPTION
**Pull request by spencer516**
For: https://github.com/machty/ember-concurrency/issues/200

### CODE REVIEW

File: addon/helpers/task.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import Ember from 'ember';
```
Comment (alexander-alvarez) [inline addon/helpers/task.js]: `import { helper } from '@ember/component/helper';`

File: addon/helpers/perform.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import Ember from 'ember';
```
Comment (alexander-alvarez) [inline addon/helpers/perform.js]: `import { helper } from '@ember/component/helper';`

File: addon/helpers/cancel-all.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import Ember from 'ember';
```
Comment (alexander-alvarez) [inline addon/helpers/cancel-all.js]: `import { helper } from '@ember/component/helper';`

**Comment by spencer516** [comment]
Happy to make that change — but to be sure — using the new module imports was [recently reverted](https://github.com/machty/ember-concurrency/commit/1dd1600a67f6d906ee327347b9dd8251a95e3810) as it caused some issues with older versions of ember. (This PR, of course, should make that no longer an issue)...but just want to be sure that change shouldn't be a separate PR.

**Comment by machty** [comment]
@spencer516 the new import style is safe so long as it's in the `addon/` folder (which it is). The pattern of keeping everything in `addon` and just re-exporting things in `app` facilitates this.

**Comment by machty** [comment]
Thank you!

**Pull request merged**